src/vue/modules/redeem-form/index.vue

```diff
@@ -0,0 +1,213 @@
+            <input-field
+              v-if="selectedSale"
+              :label="'redeem-form.invested-in' | globalize"
+              v-model="selectedSale.defaultQuoteAsset.id"
+              name="redeem-invested-in"
+              :readonly="true"
+            />
+          </div>
+        </div>
+
+        <div class="app__form-row">
+          <div class="app__form-field">
+            <input-field
+              :label="'redeem-form.total' | globalize"
+              v-model="form.totalAmount"
+              name="redeem-total"
+              :readonly="true"
+            />
+          </div>
+        </div>
+
+        <template v-if="formMixin.isConfirmationShown">
+          <form-confirmation
+            @cancel="hideConfirmation"
+            @ok="submit"
+            :is-pending="isRedeemProcessing"
+            class="app__form-confirmation"
+          />
+        </template>
+
+        <template v-else>
+          <div class="app__form-actions">
+            <button
+              v-ripple
+              type="submit"
+              class="app__form-submit-btn"
+              :disabled="formMixin.isDisabled"
+            >
+              <template>
+                {{ 'redeem-form.submit-sell-btn' | globalize }}
+              </template>
+            </button>
+          </div>
+        </template>
+      </form>
+    </div>
+    <no-data-message
+      :title="'redeem-form.no-investments' | globalize"
+      :message="'redeem-form.here-will-investments-list' | globalize"
+      v-else-if="!assetsInTheBalance.length && isInitialized"
+    />
+    <loader v-else message-id="redeem-form.loading-msg" />
+  </div>
+</template>
+
+<script>
+import { mapActions, mapMutations, mapGetters } from 'vuex'
+import { types } from './store/types'
+
+import { Wallet } from '@tokend/js-sdk'
+import { initApi } from './_api'
+
+import FormConfirmation from '@/vue/common/FormConfirmation'
+import Loader from '@/vue/common/Loader'
+import NoDataMessage from '@/vue/common/NoDataMessage'
+
+import FormMixin from '@/vue/mixins/form.mixin'
+import OfferManagerMixin from '@/vue/mixins/offer-manager.mixin'
+
+import { MathUtil } from '@/js/utils/math.util'
+
+const EVENTS = {
+  redeemed: 'redeemed',
+}
+
+export default {
+  name: 'redeem-form',
+  components: {
+    FormConfirmation,
+    NoDataMessage,
+    Loader,
+  },
+  mixins: [FormMixin, OfferManagerMixin],
+  props: {
+    wallet: {
+      type: Wallet,
+      required: true,
+    },
+    /**
+     * @property config - the config for component to use
+     * @property config.horizonURL - the url of horizon server (without version)
+     * @property config.minAmount - min allowed amount
+     * @property config.maxAmount - max allowed amount
+     */
+    config: {
+      type: Object,
+      required: true,
+    },
+  },
+  data: _ => ({
+    isInitialized: false,
+    form: {
+      asset: null,
+      quoteAsset: null,
+      totalAmount: null,
+    },
```

Comment:
do not assign `null` as default value of prop cuz you cannot be sure hen you do `form.quoteAsset.id` that it has anything already